src/cargo/util/config.rs

```diff
@@ -844,13 +841,17 @@ impl ConfigValue {
                     };
                 }
             }
-            (expected, found) => {
```

Comment:
Could a comment be added here indicating that this block is intended for "you can't switch types between tables/arrays, but you can switch other key value types"?